modules/GanneffServ.rb

```diff
@@ -100,6 +102,13 @@ def initialize()
       irc_lower($1)')
     @dbq['INCREASE_KILLS'] = DB.prepare('UPDATE ganneffserv SET kills = kills+1
       WHERE irc_lower(channel) = irc_lower($1)')
+    @dbq['INSERT_PROTECT'] = DB.prepare('INSERT INTO ganneffprotect(setter, time,
+      pattern, reason) VALUES($1, $2, $3, $4)')
+    @dbq['DELETE_PROTECT'] = DB.prepare('DELETE FROM ganneffprotect WHERE
+      irc_lower(pattern) = irc_lower($1)')
+    @dbq['GET_PROTECTED_PATTERNS'] = DB.prepare('SELECT pattern, reason FROM ganneffprotect')
+    @dbq['GET_PROTECTED_PATTERNS_DETAILED'] = DB.prepare('SELECT pattern, setter, time,
```

Comment:
`setter` here will be an ID, not a name.  Look at `GET_ALL_CHANNELS` for a simple way to get the primary nick of the setting account.